### PR TITLE
JIRA: ABC-191 Add import option to add the AlembicCurveRenderer automatically

### PR DESCRIPTION
**Pull request by clusty**
Screen Shot 2021-02-03 at 9 00 51 AM

This PR adds a new option called "**AddCurveRenderers**" which adds automatically the AlembicCurveRenderer components on every AlembicCurve.

This is useful when one desires just a visualization of the curves (or they don't have access to fancier curve rendering)

This PR affects the package user guide => to be managed through [ABC-189](https://jira.unity3d.com/browse/ABC-189).

### CODE REVIEW

File: com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs [hunk 1]
```diff
@@ -48,8 +48,13 @@ public override void OnInspectorGUI()
...
                 EditorGUILayout.PropertyField(serializedObject.FindProperty(pathSettings + "importPoints"),
                     new GUIContent("Import Points", ""));
-                EditorGUILayout.PropertyField(serializedObject.FindProperty(pathSettings + "importCurves"),
-                    new GUIContent("Import Curves", ""));
+                var importCurvesProp = serializedObject.FindProperty(pathSettings + "importCurves");
+                EditorGUILayout.PropertyField(importCurvesProp, new GUIContent("Import Curves", ""));
+                if (importCurvesProp.boolValue == true)
+                {
+                    EditorGUILayout.PropertyField(serializedObject.FindProperty(pathSettings + "createCurveRenderers"),
+                        new GUIContent("Add Curve Renderers", ""));
```
Comment (cguer) [inline com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs:56]: Suggestion for tooltip: `Automatically add "Alembic Curves Renderer" components on curve objects. This component enables basic rendering of curves for preview purposes inside the Game and Scene views.`

I think we need to mention something like "preview" or "visualize" to make sure that users don't assume a full rendering solution (which I think some will).

**Review by cguer — COMMENTED** [review]
The feature is QA approved but I think a tooltip would be needed for the option. I added a suggestion in-line, let me know what you think.

**Review by mfe — APPROVED** [review]
LGTM, but second Christophe on the tooltip.

File: com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs (hunk 1 above)
Comment (clusty) [reply com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs:56]: @sebastienduverne how does this sound?
*Automatically add "AlembicCurvesRenderer" components on curve objects. This component enables basic preview of curves.*

**Review by judubu — APPROVED** [review]
👍

File: com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs (hunk 1 above)
Comment (sebastienduverne) [reply com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs:56]: I would just change the 2nd sentence:
`This allows you to preview curves in low resolution.`

File: com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicSettings.cs [hunk 2]
```diff
@@ -174,5 +174,15 @@ public bool ImportCurves
             get { return importCurves; }
             set { importCurves = value; }
         }
+
+        [SerializeField] bool createCurveRenderers;
+        /// <summary>
+        /// If importing of Alembic curves is enabled, also create automatically the AlembicCurveRendering component.
```
Comment (sebastienduverne) [inline com.unity.formats.alembic/Runtime/Scripts/Importer/AlembicSettings.cs:180]: Suggestion:
`If you enabled the importing of Alembic curves, this method automatically creates the AlembicCurveRendering component.`

**Review by sebastienduverne — COMMENTED** [review]
See my in-line comments for changelog, tooltip and API doc.

File: com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs (hunk 1 above)
Comment (cguer) [reply com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs:56]: The preview isn't actually lower resolution. The curves will get rendered at the resolution of the scene/game view like all other objects in the scene.

Without this component though, the curves are never visible in the scene/game view. The main purpose of the component is to make the curves visible in the scene/game view but it shouldn't be used for "final quality render".

What makes this a "preview" is the limited quality/options for rendering. For example, a key feature of curves is the ability to modulate its width ([example](https://docs.arnoldrenderer.com/display/A5AFMUG/Curves#Curves-CurveWidth#Curves-CurveWidth)) but this is not possible with `AlembicCurvesRenderer` alone.

If a user wanted to control the width, they would have to code their own rendering solution (converting the curve data into visual information on screen). The demo team is also currently working on a bigger workflow (which includes Alembic as an importer) with more advanced rendering features. This is likely what users will be using in the future to get "good" curve renders inside Unity.

Comment (sebastienduverne) [reply com.unity.formats.alembic/Editor/Importer/AlembicImporterEditor.cs:56]: Thanks for the clarification. Take 2 (2nd sentence only):
`This allows you to get a basic preview of the Alembic curves in the Scene.`

As for the curve features not available, this is typically nice details to have in documentation, I'll keep it in mind.

File: com.unity.formats.alembic/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Changes in Alembic for Unity
 ## [2.2.0-exp.3] - 2021-01-22
+### Added
+- Added an option to automatically add the AlembicCurveRendering components.
```
[the remark this answers is not in the document]
Comment (sebastienduverne) [reply com.unity.formats.alembic/CHANGELOG.md:4]: New suggestion here too:
`New option to automatically add the AlembicCurveRendering components for basic preview of the curves in the Scene.`

**Review by sebastienduverne — APPROVED** [review]
LGTM

**Review by cguer — APPROVED** [review]
QA approved 👍!

**Pull request merged**